yjit/src/codegen.rs

```diff
@@ -341,24 +341,36 @@ fn gen_save_sp_with_offset(asm: &mut Assembler, offset: i8) {
     }
 }
 
-/// jit_save_pc() + gen_save_sp(). Should be used before calling a routine that
-/// could:
+/// jit_save_pc() + gen_save_sp(). Should be used before calling a routine that could:
 ///  - Perform GC allocation
 ///  - Take the VM lock through RB_VM_LOCK_ENTER()
 ///  - Perform Ruby method call
+///
+/// If the routine doesn't call arbitrary methods, use jit_prepare_allocation() instead.
 fn jit_prepare_routine_call(
```

Comment:
Just to improve the clarity about what this function does, we could rename it as follows?

```suggestion
fn jit_prepare_call_non_leaf(
```

That's the crux of it, right, that this can indirectly call binding because the C function can call back into Ruby code?

Reply:
Right. If the C function calls methods, all bets are off. Calling into blocks is probably more common than using binding, though.

Reply:
I changed it to `jit_prepare_non_leaf_call` https://github.com/ruby/ruby/pull/9775/commits/a2ddbf3b354b7d16e709e0e963cd7445772c5dd3 since "prepare xxx call" like `jit_prepare_routine_call` seemed more natural to me. WDYT?